tests/test_netcdf_data.py

```diff
@@ -48,6 +49,36 @@ def test_enter_nc_files_list(self):
         self.assertIsInstance(nc_data.dataset, xarray.Dataset)
         self.assertTrue(nc_data._dataset_open)
 
+    def test_enter_zarr_file(self):
+        ds = xarray.Dataset({
+            "votemper": xarray.DataArray(
+                            data   = numpy.random.rand(4,4,4,4),   
+                            dims   = ['depth', 'latitude', 'longitude', 'time'],
+                
+                            coords = { "depth": (["depth"], [4.94025e-01, 1.54138e+00, 2.64567e+00, 3.81949e+00]),
+                                    "latitude": (["latitude"], [-80. , -79.8, 89.6,  89.8]),
+                                    "longitude": (["longitude"], [2.000e-01, 4.000e-01, 3.594e+02, 3.598e+02]),
+                                    "time": (["time"], [2.211494e+09, 2.211581e+09, 2.211667e+09, 2.211754e+09]),
+                                },
+                            attrs  = {
+                                "units" : "Kelvin",
+                                "long_name" : "Sea water potential temperature",
+                                "valid_min" : 173.0,
+                                "valid_max" : 373.0,
+                                }
+                            ),
+                    }
+        )
+    
+        if not (os.path.exists("tests/testdata/giops_test.zarr")):
+            ds.to_zarr("tests/testdata/giops_test.zarr")
```

Comment:
Thanks for creating a test data on the fly.  But is the the recommended way of creating test data in the testing framework? Do we need to clean up this file after the test? What about the other test_zarr_* tests --- do we not need to create the test file first for those as well?

There should be someway of doing up a "setup" and "teardown" within the testing framework.

Reply:
I don't see any conflicts with other test methods by creating test data in the testing framework.
The latest commit won't require us to clean the data file, it creates and delete the testing data after the tests.